### PR TITLE
gh-103272: add regression test for getattr that raises

### PR DESCRIPTION
**Pull request by sunmy2019**
Add test case taken from gh-103272 for #103332.

### CODE REVIEW

**Review by hauntsaninja — COMMENTED** [review]
Awesome work, thanks for narrowing this down!

I think we can make the test case slightly simpler to make it clearer what's getting tested:
```
class A:
    def __getattr__(self, name):
        raise ValueError
    @property
    def foo(self):
        return self.__getattr__("asdf")

A().foo
```
And maybe rename the test to `test_getattr_raises` or something

**Comment by sobolevn** [comment]
I've got the revert merged, now this test must pass :)

File: Lib/test/test_descr.py [hunk 1]
```diff
@@ -5003,6 +5004,26 @@ class Child(Parent):
         gc.collect()
         self.assertEqual(Parent.__subclasses__(), [])
 
+    def test_getattribute(self):
+        # add test case for gh-103272
+        class A:
+            def __init__(self) -> None:
+                self.bar = 0
+
+            def __getattribute__(self, name):
```
Comment (sobolevn) [inline Lib/test/test_descr.py]: Yes, I agree with @hauntsaninja, this does not seem required. Or is it?

File: Misc/NEWS.d/next/Tests/2023-04-07-07-56-39.gh-issue-103272.QosA_L.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Add test case for regressions caused by gh-102213.
```
Comment (sobolevn) [inline Misc/NEWS.d/next/Tests/2023-04-07-07-56-39.gh-issue-103272.QosA_L.rst]: We don't add news for test changes :)

**Review by sobolevn — COMMENTED** [review]
Thank you!

File: Lib/test/test_descr.py (hunk 1 above)
Comment (sunmy2019) [reply Lib/test/test_descr.py]: It isn't required. I will simplify it

File: Lib/test/test_descr.py [hunk 3]
```diff
@@ -5003,6 +5003,19 @@ class Child(Parent):
         gc.collect()
         self.assertEqual(Parent.__subclasses__(), [])
 
+    def test_attr_raise_through_property(self):
+        # add test case for gh-103272
+        class A:
+            def __getattr__(self, name):
+                raise ValueError
```
Comment (sobolevn) [inline Lib/test/test_descr.py]: One last nit: I think we can use an error message here and check it in `assertRaisesRegex`. Right now, technically some different `ValueError` might fit :(

**Review by sobolevn — APPROVED** [review]
Congrats on your first CPython PR 🎉

**Review by hauntsaninja — APPROVED** [review]
Thanks again, this is great!

**Pull request merged**